pkg/azure/actuator_test.go

```diff
@@ -261,196 +269,281 @@ func TestActuator(t *testing.T) {
 			mockAppClient: func(mockCtrl *gomock.Controller) *azuremock.MockAppClient {
 				client := azuremock.NewMockAppClient(mockCtrl)
 				client.EXPECT().List(gomock.Any(), gomock.Any()).Return(
-					[]graphrbac.Application{testAADApplication()},
-					nil,
+					[]graphrbac.Application{testAADApplication()}, nil,
 				)
 				client.EXPECT().Delete(gomock.Any(), testAppRegObjID)
-
 				return client
 			},
-			mockRoleAssignmentsClient:  mockRoleAssignmentClientNoCalls,
-			mockServicePrincipalClient: mockServicePrincipalClientNoCalls,
+		},
+		{
+			name: "Failed to cleanup",
+			expectedErr: &actuatoriface.ActuatorError{
+				ErrReason: minterv1.OrphanedCloudResource,
+				Message:   fmt.Sprintf("unable to clean up App Registration / Service Principal: %s", testAppRegName),
+			},
+			existing: func() []runtime.Object {
+				objects := defaultExistingObjects()
+
+				// Add the existing targetSecret since we are mocking up
+				// a previously migrated scenario.
+				targetSecret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      testTargetSecretName,
+						Namespace: testTargetSecretNamespace,
+					},
+					Data: map[string][]byte{
+						azure.AzureClientID:       []byte(mintedClientID),
+						azure.AzureClientSecret:   []byte(mintedClientSecret),
+						azure.AzureRegion:         []byte(rootRegion),
+						azure.AzureResourceGroup:  []byte(rootResourceGroup),
+						azure.AzureResourcePrefix: []byte(rootResourcePrefix),
+						azure.AzureSubscriptionID: []byte(rootSubscriptionID),
+						azure.AzureTenantID:       []byte(rootTenantID),
+					},
+				}
+
+				objects = append(objects, targetSecret)
+
+				return objects
+			}(),
+			credentialsRequest: func() *minterv1.CredentialsRequest {
+				// Create a credreq that resembles what one previously handled via mint mode
+				// would look like.
+				cr := testCredentialsRequest(t)
+
+				rawStatus := &minterv1.AzureProviderStatus{
+					ServicePrincipalName: testAppRegName,
+					AppID:                testAppRegID,
+				}
+				encodedStatus, err := codec.EncodeProviderStatus(rawStatus)
+				require.NoError(t, err, "error encoding status")
+
+				cr.Status.ProviderStatus = encodedStatus
+				cr.Status.Provisioned = true
+
+				return cr
+			}(),
 			op: func(actuator *azure.Actuator, cr *minterv1.CredentialsRequest) error {
-				return actuator.Delete(context.TODO(), cr)
+				return actuator.Update(context.TODO(), cr)
 			},
 			validate: func(t *testing.T, c client.Client) {
 				cr := getCredRequest(t, c)
-				s := &corev1.Secret{}
-				// secret should be deleted
-				assert.Error(t, c.Get(context.TODO(),
-					types.NamespacedName{Name: cr.Spec.SecretRef.Name, Namespace: cr.Spec.SecretRef.Namespace},
-					s),
+
+				targetSecret := getCredRequestTargetSecret(t, c, cr)
+
+				rootSecret := getRootSecret(t, c)
+
+				// The targetSecret should be a copy of the root secret.
+				assertSecretEquality(t, rootSecret, targetSecret)
+			},
```

Comment:
Do we need to check for `OrphanedCloudResource` condition here?

Reply:
The condition is set by the credentials_request controller, so the best we can do with the test at the actuator-level is check for the right error returned by the actuator. (Although I should go ahead and add a test for the condition in the credentials_request controller. I'll work on that now).